# Before we start

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
pd.set_option('display.width', 180)


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
from google.colab import output
from IPython.display import Javascript

def resize_colab_cell():
    display(Javascript('google.colab.output.setIframeHeight(0, true, {maxHeight: 5000})'))

get_ipython().events.register('pre_run_cell', resize_colab_cell)


<IPython.core.display.Javascript object>

In [ ]:
import pandas as pd

# Replace 'path/to/homelessness.csv' with the path to your CSV file
df = pd.read_csv('/content/drive/MyDrive/WK10-pandas/Resources/data.csv', index_col = 0)

# Print the first 5 rows of the DataFrame
print(df.head())


          Pulse  Maxpulse  Calories
Duration                           
60          110       130     409.1
60          117       145     479.0
60          103       135     340.0
45          109       175     282.4
45          117       148     406.0


index_col is a parameter of the pandas.read_csv function that specifies which column of the CSV file to use as the row labels of the DataFrame1. If index_col is set to 0, it means that the first column of the CSV file will be used as the row labels. For example, if the CSV file looks like this:

As we do not need to do it for our data so we shall remove the index_col parameter but we shall need to use it later.

* * *

#Viewing the Data
----------------

One of the most used method for getting a quick overview of the DataFrame, is the `head()` method.

The `head()` method returns the headers and a specified number of rows, starting from the top.





In [ ]:
import pandas as pd

# Replace 'path/to/homelessness.csv' with the path to your CSV file
df = pd.read_csv('/content/drive/MyDrive/WK10-pandas/Resources/data.csv')

# Print the first 5 rows of the DataFrame
print(df.head(10))

   Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.0
6        60    110       136     374.0
7        45    104       134     253.3
8        30    109       133     195.1
9        60     98       124     269.0


In our examples we will be using a CSV file called 'data.csv'.

Download [data.csv](https://www.w3schools.com/python/pandas/data.csv), or open [data.csv](https://www.w3schools.com/python/pandas/data.csv.txt) in your browser.

**Note:** if the number of rows is not specified, the `head()` method will return the top 5 rows.

### Example

Print the first 5 rows of the DataFrame:
```
import pandas as pd

# Replace 'path/to/homelessness.csv' with the path to your CSV file
df = pd.read_csv('/content/drive/MyDrive/WK10-pandas/Resources/data.csv')

# Print the first 5 rows of the DataFrame
print(df.head())
```
[Try it Yourself »](https://www.w3schools.com/python/pandas/trypandas.asp?filename=demo_pandas_head)

* * *





In [ ]:
import pandas as pd

# Replace 'path/to/homelessness.csv' with the path to your CSV file
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/WK10-Pandas Learner/Resources/data.csv')

# Print the first 5 rows of the DataFrame
print(df.head())

FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/Colab Notebooks/WK10-Pandas Learner/Resources/data.csv'

There is also a `tail()` method for viewing the _last_ rows of the DataFrame.

The `tail()` method returns the headers and a specified number of rows, starting from the bottom.

* * *

In [ ]:
print(df.tail())

     Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4


Info About the Data
-------------------

The DataFrames object has a method called `info()`, that gives you more information about the data set.

### Example
```
#Print information about the data:

print(df.info())
```


In [ ]:
#Print information about the data:
import pandas as pd

# Replace 'path/to/homelessness.csv' with the path to your CSV file
df = pd.read_csv('/content/drive/MyDrive/WK10-pandas/Resources/data.csv')
print(df.info())

NameError: name 'df' is not defined

Result
------
```
  <class 'pandas.core.frame.DataFrame'>
  RangeIndex: 169 entries, 0 to 168
  Data columns (total 4 columns):
   #   Column    Non-Null Count  Dtype  
  ---  ------    --------------  -----  
   0   Duration  169 non-null    int64  
   1   Pulse     169 non-null    int64  
   2   Maxpulse  169 non-null    int64  
   3   Calories  164 non-null    float64
  dtypes: float64(1), int64(3)
  memory usage: 5.4 KB
  None
  ```  

[Try it Yourself »](https://www.w3schools.com/python/pandas/trypandas.asp?filename=demo_pandas_info)

Result Explained
----------------

The result tells us there are 169 rows and 4 columns:
```
  RangeIndex: 169 entries, 0 to 168
  Data columns (total 4 columns):
```
And the name of each column, with the data type:
```
   #   Column    Non-Null Count  Dtype  
  ---  ------    --------------  -----  
   0   Duration  169 non-null    int64  
   1   Pulse     169 non-null    int64  
   2   Maxpulse  169 non-null    int64  
   3   Calories  164 non-null    float64
```
Null Values
-----------

The `info()` method also tells us how many Non-Null values there are present in each column, and in our data set it seems like there are 164 of 169 Non-Null values in the "Calories" column.

Which means that there are 5 rows with no value at all, in the "Calories" column, for whatever reason.

Empty values, or Null values, can be bad when analyzing data, and you should consider removing rows with empty values. This is a step towards what is called _cleaning data_, and you will learn more about that in the next chapters.

* * *